From d3ee6e4630f0f7d76099cb1c12a339074d3f2992 Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Sat, 16 May 2026 13:10:58 +0800
Subject: [PATCH 5/5] Namespace ZXING_* macros

Avoids clashing with those from ZXing itself, especially when we have to
modify them like for the detection of the upcoming ZXing 3.0.
---
 src/lib/barcodedecoder.cpp        | 24 ++++++++++++------------
 src/lib/config-kitinerary.h.cmake | 16 ++++++++--------
 src/lib/extractorcapabilities.cpp |  2 +-
 src/lib/qimagepurebinarizer.cpp   |  4 ++--
 src/lib/qimagepurebinarizer_p.h   |  6 +++---
 5 files changed, 26 insertions(+), 26 deletions(-)

diff --git a/src/lib/barcodedecoder.cpp b/src/lib/barcodedecoder.cpp
index ac2058d3..a49d7034 100644
--- a/src/lib/barcodedecoder.cpp
+++ b/src/lib/barcodedecoder.cpp
@@ -15,7 +15,7 @@
 #include <QString>
 
 #define ZX_USE_UTF8 1
-#if ZXING_USE_READBARCODE
+#if KZXING_USE_READBARCODE
 #include <ZXing/ReadBarcode.h>
 #else
 #include <ZXing/DecodeHints.h>
@@ -133,7 +133,7 @@ struct {
     BarcodeDecoder::BarcodeType type;
     ZXing::BarcodeFormat zxingType;
 } static constexpr const zxing_format_map[] = {
-#if ZXING_VERSION > QT_VERSION_CHECK(1, 1, 1)
+#if KZXING_VERSION > QT_VERSION_CHECK(1, 1, 1)
     { BarcodeDecoder::Aztec, ZXing::BarcodeFormat::Aztec },
     { BarcodeDecoder::QRCode, ZXing::BarcodeFormat::QRCode },
     { BarcodeDecoder::PDF417, ZXing::BarcodeFormat::PDF417 },
@@ -154,7 +154,7 @@ struct {
 
 static ZXing::BarcodeFormats typeToFormats(BarcodeDecoder::BarcodeTypes types)
 {
-#if ZXING_VERSION < QT_VERSION_CHECK(3, 0, 0)
+#if KZXING_VERSION < QT_VERSION_CHECK(3, 0, 0)
     ZXing::BarcodeFormats formats;
 #else
     std::vector<ZXing::BarcodeFormat> formats;
@@ -162,7 +162,7 @@ static ZXing::BarcodeFormats typeToFormats(BarcodeDecoder::BarcodeTypes types)
 
     for (auto i : zxing_format_map) {
         if (types & i.type) {
-#if ZXING_VERSION < QT_VERSION_CHECK(3, 0, 0)
+#if KZXING_VERSION < QT_VERSION_CHECK(3, 0, 0)
             formats |= i.zxingType;
 #else
             formats.push_back(i.zxingType);
@@ -182,14 +182,14 @@ BarcodeDecoder::BarcodeType formatToType(ZXing::BarcodeFormat format)
     return BarcodeDecoder::None;
 }
 
-#if ZXING_USE_READBARCODE
+#if KZXING_USE_READBARCODE
 static ZXing::ImageFormat zxingImageFormat(QImage::Format format)
 {
     switch (format) {
         case QImage::Format_ARGB32:
         case QImage::Format_RGB32:
 #if Q_BYTE_ORDER == Q_LITTLE_ENDIAN
-#if ZXING_VERSION >= QT_VERSION_CHECK(2, 3, 0)
+#if KZXING_VERSION >= QT_VERSION_CHECK(2, 3, 0)
             return ZXing::ImageFormat::RGBA;
 #else
             return ZXing::ImageFormat::RGBX;
@@ -201,7 +201,7 @@ static ZXing::ImageFormat zxingImageFormat(QImage::Format format)
             return ZXing::ImageFormat::RGB;
         case QImage::Format_RGBX8888:
         case QImage::Format_RGBA8888:
-#if ZXING_VERSION >= QT_VERSION_CHECK(2, 3, 0)
+#if KZXING_VERSION >= QT_VERSION_CHECK(2, 3, 0)
             return ZXing::ImageFormat::RGBA;
 #else
             return ZXing::ImageFormat::RGBX;
@@ -213,7 +213,7 @@ static ZXing::ImageFormat zxingImageFormat(QImage::Format format)
     }
     Q_UNREACHABLE();
 }
-#if ZXING_VERSION >= QT_VERSION_CHECK(3, 0, 0)
+#if KZXING_VERSION >= QT_VERSION_CHECK(3, 0, 0)
 static ZXing::Barcode zxingReadBarcode(const QImage &img, const ZXing::ReaderOptions &hints)
 #else
 static ZXing::Result zxingReadBarcode(const QImage &img, const ZXing::DecodeHints &hints)
@@ -225,18 +225,18 @@ static ZXing::Result zxingReadBarcode(const QImage &img, const ZXing::DecodeHint
 
 void BarcodeDecoder::decodeZxing(const QImage &img, BarcodeDecoder::BarcodeTypes format, BarcodeDecoder::Result &result) const
 {
-#if ZXING_VERSION >= QT_VERSION_CHECK(3, 0, 0)
+#if KZXING_VERSION >= QT_VERSION_CHECK(3, 0, 0)
     ZXing::ReaderOptions hints;
 #else
     ZXing::DecodeHints hints;
 #endif
-#if ZXING_VERSION >= QT_VERSION_CHECK(1, 1, 0)
+#if KZXING_VERSION >= QT_VERSION_CHECK(1, 1, 0)
     hints.setFormats(typeToFormats(format));
 #else
     hints.setPossibleFormats(typeToFormats(format));
 #endif
 
-#if ZXING_USE_READBARCODE
+#if KZXING_USE_READBARCODE
     hints.setBinarizer(ZXing::Binarizer::FixedThreshold);
     hints.setIsPure((format & BarcodeDecoder::IgnoreAspectRatio) == 0);
 
@@ -250,7 +250,7 @@ void BarcodeDecoder::decodeZxing(const QImage &img, BarcodeDecoder::BarcodeTypes
 #endif
 
     if (res.isValid()) {
-#if ZXING_VERSION >= QT_VERSION_CHECK(1, 4, 0)
+#if KZXING_VERSION >= QT_VERSION_CHECK(1, 4, 0)
         // detect content type
         std::string zxUtf8Text;
         if (res.contentType() == ZXing::ContentType::Text) {
diff --git a/src/lib/config-kitinerary.h.cmake b/src/lib/config-kitinerary.h.cmake
index 0a7a621d..dd709902 100644
--- a/src/lib/config-kitinerary.h.cmake
+++ b/src/lib/config-kitinerary.h.cmake
@@ -14,20 +14,20 @@
 #define KPOPPLER_VERSION_PATCH @POPPLER_VERSION_PATCH@
 #define KPOPPLER_VERSION ((@POPPLER_VERSION_MAJOR@<<16)|(@POPPLER_VERSION_MINOR@<<8)|(@POPPLER_VERSION_PATCH@))
 
-#define ZXING_VERSION_STRING "@ZXing_VERSION@"
-#define ZXING_VERSION_MAJOR @ZXing_VERSION_MAJOR@
-#define ZXING_VERSION_MINOR @ZXing_VERSION_MINOR@
-#define ZXING_VERSION_PATCH @ZXing_VERSION_PATCH@
-#define ZXING_VERSION ((@ZXing_VERSION_MAJOR@<<16)|(@ZXing_VERSION_MINOR@<<8)|(@ZXing_VERSION_PATCH@))
+#define KZXING_VERSION_STRING "@ZXing_VERSION@"
+#define KZXING_VERSION_MAJOR @ZXing_VERSION_MAJOR@
+#define KZXING_VERSION_MINOR @ZXing_VERSION_MINOR@
+#define KZXING_VERSION_PATCH @ZXing_VERSION_PATCH@
+#define KZXING_VERSION ((@ZXing_VERSION_MAJOR@<<16)|(@ZXing_VERSION_MINOR@<<8)|(@ZXing_VERSION_PATCH@))
 
 // QT_VERSION_CHECK isn't available in here for the below check
 #define K_VERSION_CHECK(major, minor, patch) ((major<<16)|(minor<<8)|(patch))
 
 // this might compile with older versions too, but it only actually works post 1.1.1
-#if ZXING_VERSION > K_VERSION_CHECK(1, 1, 1)
-    #define ZXING_USE_READBARCODE 1
+#if KZXING_VERSION > K_VERSION_CHECK(1, 1, 1)
+    #define KZXING_USE_READBARCODE 1
 #else
-    #define ZXING_USE_READBARCODE 0
+    #define KZXING_USE_READBARCODE 0
 #endif
 
 #cmakedefine01 HAVE_LIBXML2
diff --git a/src/lib/extractorcapabilities.cpp b/src/lib/extractorcapabilities.cpp
index 845c925c..867cd846 100644
--- a/src/lib/extractorcapabilities.cpp
+++ b/src/lib/extractorcapabilities.cpp
@@ -35,7 +35,7 @@ QString ExtractorCapabilities::capabilitiesString()
 
         "iCal support        : kcal (" KCALENDARCORE_VERSION_STRING ")\n"
 
-        "Barcode decoder     : ZXing (" ZXING_VERSION_STRING ")\n"
+        "Barcode decoder     : ZXing (" KZXING_VERSION_STRING ")\n"
 
         "Phone number decoder: "
 #if HAVE_PHONENUMBER
diff --git a/src/lib/qimagepurebinarizer.cpp b/src/lib/qimagepurebinarizer.cpp
index a5863009..a632a8af 100644
--- a/src/lib/qimagepurebinarizer.cpp
+++ b/src/lib/qimagepurebinarizer.cpp
@@ -5,7 +5,7 @@
 */
 
 #include "qimagepurebinarizer_p.h"
-#if !ZXING_USE_READBARCODE
+#if !KZXING_USE_READBARCODE
 
 #include <ZXing/BitArray.h>
 
@@ -59,7 +59,7 @@ bool QImagePureBinarizer::getBlackRow(int y, ZXing::BitArray &row) const
     return true;
 }
 
-#if ZXING_VERSION >= QT_VERSION_CHECK(1, 1, 1)
+#if KZXING_VERSION >= QT_VERSION_CHECK(1, 1, 1)
 bool QImagePureBinarizer::getPatternRow(int y, ZXing::PatternRow& res) const
 {
     res.clear();
diff --git a/src/lib/qimagepurebinarizer_p.h b/src/lib/qimagepurebinarizer_p.h
index 255a9c1c..7cead76c 100644
--- a/src/lib/qimagepurebinarizer_p.h
+++ b/src/lib/qimagepurebinarizer_p.h
@@ -7,7 +7,7 @@
 #pragma once
 
 #include "config-kitinerary.h"
-#if !ZXING_USE_READBARCODE
+#if !KZXING_USE_READBARCODE
 
 #include <QImage>
 
@@ -31,12 +31,12 @@ public:
     bool isPureBarcode() const override;
 
     bool getBlackRow(int y, ZXing::BitArray &row) const
-#if ZXING_VERSION < QT_VERSION_CHECK(1, 1, 1)
+#if KZXING_VERSION < QT_VERSION_CHECK(1, 1, 1)
     // no longer present in later versions of 1.1.1...
     override
 #endif
     ;
-#if ZXING_VERSION >= QT_VERSION_CHECK(1, 1, 1)
+#if KZXING_VERSION >= QT_VERSION_CHECK(1, 1, 1)
     bool getPatternRow(int y, ZXing::PatternRow& res) const override;
 #endif
     std::shared_ptr<const ZXing::BitMatrix> getBlackMatrix() const override;
-- 
2.52.0

